packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -28,38 +68,93 @@ export class State {
    */
   public readonly stateName: string;
 
+  private readonly transitionEvents: TransitionEvent[] = [];
+
   constructor(private readonly props: StateProps) {
     this.stateName = props.stateName;
   }
 
   /**
-   * Return the state property JSON.
+   * Add a transition event to the state.
+   * The transition event will be triggered if condition is evaluated to TRUE.
+   *
+   * @param targetState the state that will be transit to when the event triggered
+   * @param options transition options including the condition that causes the state transition
+   */
+  public transitionTo(targetState: State, options: TransitionOptions) {
+    const alreadyAdded = this.transitionEvents.some((event) => event.nextState === targetState);
```

Comment:
The parentheses are redundant here, and this is a `TransitionEvent`, not an `Event`, so I would actually make that explicit:

```suggestion
    const alreadyAdded = this.transitionEvents.some(transitionEvent => transitionEvent.nextState === targetState);
```